### DISCUSSION TITLE
Possibility to omit/hide a field label

### DISCUSSION THREAD

**Discussion by piljac1** [laravel/nova-issues#4298] [Ideas & Feature Requests]
## Idea
It would be great if you would be able to omit/hide a field label from forms and detail views.

## Why?
For example, if you have a separate field panel that has a single BooleanGroup listing categories provided by an API, it would be nice to avoid output "Categories" labels twice (name of the panel + field label). The name of the panel is enough to understand what we're outputting (in the case of a single field). No need to repeat it on the field label.

Right now, it is possible to have an empty label, but in the case of a single field, it is just really weird looking.

**Comment by Anticom** [comment]
@piljac1 Dunno when this was introduced, but in Nova `5.1.7` `Laravel\Nova\Fields\Field` has a public `$withLabel` property that seems to control whether Nova's frontend is showing a label on the field or not. However there's no (fluent) setter for that property making it hard to configure. I suppose it's intended for more specialized Fields extending from it.

> [!WARNING]  
> Since this is not documented anywhere I would consider it internal meaning the package maintainers may change / remove that property at any time. Also as stated above there's no convenient way to configure it on a per field basis.  
> Also for the Field I tested it had some layout issues in combination with `fullWidth()`.

---

You could work around that limitation by extending a given field and adding a fluent setter to it and use that one.

For example:
```php
<?php

namespace App\Nova\Fields;

use Laravel\Nova\Fields\Text as NovaText;

class Text extends NovaText
{
    public function withoutLabel()
    {
        $this->withLabel = false;

        return $this;
    }
}
```


I have no clue, why they did not expose that property via a setter but I'm sure they have their reasons.

**Reply by crynobone** [reply]
This property is designed for Repeater Field and not documentation for other usage. Use with caution as future release (related to Repeater might break this).

**Reply by osbre** [reply]
Or via macro:

```php
use Laravel\Nova\Fields\Field;

// In NovaServiceProvider's boot method:

Field::macro('withoutLabel', function () {
    $this->withLabel = false;

    return $this;
});
```
